include/signature.h

```diff
@@ -276,3 +276,14 @@ GBytes *cms_encrypt(GBytes *content, gchar **recipients, GError **error);
  * @return decrypted CMS data, NULL if failed
  */
 GBytes *cms_decrypt(GBytes *content, const gchar *certfile, const gchar *keyfile, GError **error);
+
+
+/**
+ * Dump content of enevloped CMS.
+ *
+ * @param[in] sig enveloped CMS data to dump
+ * @param[out] error return location for a GError, or NULL
+ *
+ * @return string representation for printing, or NULL on error
+ */
+gchar* envelopeddata_to_string(GBytes *sig, GError **error);
```

Comment:
This doesn't seem print the enveloped data (which would be the encrypted payload), but the "envelope data" (i.e. the metadata).

Reply:
The CMS contentType is `pkcs7-envelopedData`.

If you dump it, it will look like:

```
CMS_ContentInfo: 
  contentType: pkcs7-envelopedData (1.2.840.113549.1.7.3)
  d.envelopedData: 
    version: 2
    originatorInfo: <ABSENT>
    recipientInfos:
      d.kari: 
        version: 3
        d.originatorKey: 
          algorithm: 
            algorithm: id-ecPublicKey (1.2.840.10045.2.1)
            parameter: <ABSENT>
          publicKey:  (0 unused bits)
            0000 - 04 10 14 e6 c6 bc be 84-79 a2 2c 98 37 c1   ........y.,.7.
            000e - d1 ba 99 e0 87 61 8d 6b-a4 5e bb e6 db 60   .....a.k.^...`
            001c - 16 b4 e9 a4 5b 4d a9 74-2b 15 19 7e 16 bc   ....[M.t+..~..
            002a - a1 0c 00 c1 af 1b c5 a4-80 d9 ba 7c 52 c1   ...........|R.
            0038 - d5 89 33 41 21 c2 3c 04-d3                  ..3A!.<..
        ukm: <ABSENT>
        keyEncryptionAlgorithm: 
          algorithm: dhSinglePass-stdDH-sha1kdf-scheme (1.3.133.16.840.63.0.2)
          parameter: SEQUENCE:
    0:d=0  hl=2 l=  11 cons: SEQUENCE          
    2:d=1  hl=2 l=   9 prim:  OBJECT            :id-aes256-wrap
        recipientEncryptedKeys:
            d.issuerAndSerialNumber: 
              issuer: O=Test Org, CN=Test Org Development 00008
              serialNumber: 0x16D1AA713603E287808FB18E96F9AFC06BB81E1C
            encryptedKey: 
              0000 - da 66 14 d4 a3 1c 4f ca-c8 a6 b9 fc c5 42   .f....O......B
              000e - 20 7e b4 39 97 25 55 9f-fa 5d c1 19 db e1    ~.9.%U..]....
              001c - d0 00 96 e1 cb 00 ad f0-31 8e 47 8f         ........1.G.
```

The 'encryptedContent:' in turn is part of the 'envelopedData'

We print what is inside the 'envelopedData'. Thus I would say the naming fits, doesn't it?

Reply:
OK, you convinced me. :)